projects/core/src/lib/project/project.service.ts

```diff
@@ -874,24 +874,27 @@ export class ProjectService {
 
     /**
      * Create a stream that signals whether a running query should be aborted because the results are no longer needed.
-     * It takes the current zoomLevel and extent of a tile at the time of querying as a parameter in order to determine
-     * whether a change on the map view makes the results obsolete.
+     * It takes the layerId, current zoomLevel and extent of a tile at the time of querying as a parameter in order to
+     * determine whether a change in the layer list or on the map view makes the results obsolete.
      */
-    createQueryAbortStream(tileZoomLevel: number, tileExtent: Extent): Observable<void> {
+    createQueryAbortStream(layerId: number, tileZoomLevel: number, tileExtent: Extent): Observable<void> {
         const tileResolution = this.mapService.getView().getResolutionForZoom(tileZoomLevel);
 
         const observables: Array<Observable<any>> = [
             this.getTimeStream(),
             this.mapService.getViewportSizeStream(),
             this.userService.getSessionTokenForRequest(),
             this.getSpatialReferenceStream(),
+            this.getLayerStream(),
```

Comment:
Woudln't it be enough to look at only one layer in `layers`?

Reply:
is there a way to monitor a single layer? If there is, what is it?

Reply:
Yeah, the entries/oversables in the `layers` variable. It is set in `layerChangesStream` (or similar) and completed in `removeLayerSubscriptions`.

Reply:
ok, but what is your idea here? I would need an observable for each layer. Like `layerDeletion$: Map<number, Subscription>()` and subscribe to it. Is it really worth to introduce this overhead, instead of just scanning the existing layers? it's not like we are removing hundreds of layers all the time.

If that's not what you meant, please give me a more detailed description of your desired solution.

Reply:
5b0f8bf4f0801cb28cc37b4a8dc339c3eea94bb5